artifactory/services/federatedrepository.go

```diff
@@ -0,0 +1,448 @@
+}
+
+func NewDebianFederatedRepositoryParams() DebianFederatedRepositoryParams {
+	return DebianFederatedRepositoryParams{FederatedRepositoryBaseParams: FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "debian"}}
+}
+
+type DockerFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+	MaxUniqueTags       int    `json:"maxUniqueTags,omitempty"`
+	DockerApiVersion    string `json:"dockerApiVersion,omitempty"`
+	BlockPushingSchema1 *bool  `json:"blockPushingSchema1,omitempty"`
+	DockerTagRetention  int    `json:"dockerTagRetention,omitempty"`
+}
+
+func NewDockerFederatedRepositoryParams() DockerFederatedRepositoryParams {
+	return DockerFederatedRepositoryParams{FederatedRepositoryBaseParams: FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "docker"}}
+}
+
+type GemsFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+}
+
+func NewGemsFederatedRepositoryParams() GemsFederatedRepositoryParams {
+	return GemsFederatedRepositoryParams{FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "gems"}}
+}
+
+type GenericFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+}
+
+func NewGenericFederatedRepositoryParams() GenericFederatedRepositoryParams {
+	return GenericFederatedRepositoryParams{FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "generic"}}
+}
+
+type GitlfsFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+}
+
+func NewGitlfsFederatedRepositoryParams() GitlfsFederatedRepositoryParams {
+	return GitlfsFederatedRepositoryParams{FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "gitlfs"}}
+}
+
+type GoFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+}
+
+func NewGoFederatedRepositoryParams() GoFederatedRepositoryParams {
+	return GoFederatedRepositoryParams{FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "go"}}
+}
+
+type GradleFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+	CommonMavenGradleFederatedRepositoryParams
+}
+
+func NewGradleFederatedRepositoryParams() GradleFederatedRepositoryParams {
+	return GradleFederatedRepositoryParams{FederatedRepositoryBaseParams: FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "gradle"}}
+}
+
+type HelmFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+}
+
+func NewHelmFederatedRepositoryParams() HelmFederatedRepositoryParams {
+	return HelmFederatedRepositoryParams{FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "helm"}}
+}
+
+type IvyFederatedRepositoryParams struct {
+	FederatedRepositoryBaseParams
+	CommonMavenGradleFederatedRepositoryParams
+}
+
+func NewIvyFederatedRepositoryParams() IvyFederatedRepositoryParams {
+	return IvyFederatedRepositoryParams{FederatedRepositoryBaseParams: FederatedRepositoryBaseParams{Rclass: "federated", PackageType: "ivy"}}
+}
+
+type CommonMavenGradleFederatedRepositoryParams struct {
```

Comment:
@eyalbe4 there are common properties that apply to `maven`, `gradle`, `ivy`, and `sbt` and I'm struggling to come up with a good name instead of `CommonMavenGradleIvySbtRepositoryParams` but I'm unsure what name could be used to define them all. Do you have any ideas or are you happy to just keep adding the package names to the struct name?

Reply:
It's definitely not a perfect name, but the only name I can come up with right now is ```JavaPackageManagersRepositoryParams```.

Reply:
Done ✔️